integration-tests/rpc_integration_test.go

```diff
@@ -0,0 +1,465 @@
+
+	expectedGetStatsResponse := &rpc.GetStatsResponse{
+		Version:              "development",
+		PubSubTopic:          "/0x-orders/network/1337/version/1",
+		Rendezvous:           "/0x-mesh/network/1337/version/1",
+		PeerID:               jsonLog["myPeerID"].(string),
+		EthereumChainID:      1337,
+		LatestBlock:          rpc.LatestBlock{},
+		NumOrders:            0,
+		NumPeers:             0,
+		MaxExpirationTime:    "115792089237316195423570985008687907853269984665640564039457584007913129639935",
+		StartOfCurrentUTCDay: getUTCMidnightOfDate(time.Now()),
+	}
+
+	client, err := rpc.NewClient(standaloneRPCEndpoint + strconv.Itoa(rpcPort+count))
+	require.NoError(t, err)
+
+	getStatsResponse, err := client.GetStats()
+
+	// HACK(jalextowle): Zeroing the Number and Hash fields of LatestBlock
+	//                   allows us to test more of the "GetStats" response
+	//                   without being too restrictive about the blockchain
+	//                   that is being used.
+	getStatsResponse.LatestBlock = rpc.LatestBlock{}
+
+	require.NoError(t, err)
+	require.Equal(t, expectedGetStatsResponse, getStatsResponse)
+
+	// Cancel the context and wait for all outstanding goroutines to finish.
+	cancel()
+	wg.Wait()
+}
+
+func TestOrdersSubscription(t *testing.T) {
+	teardownSubTest := setupSubTest(t)
+	defer teardownSubTest(t)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// Remove the old database and p2p files.
+	removeOldFiles(t, ctx)
+
+	buildStandaloneForTests(t, ctx)
+
+	logMessages := make(chan string, 1024)
+
+	count := int(atomic.AddInt32(nodeCount, 1))
+
+	// Set up the dummy handler with an addOrdersHandler
+	wg := &sync.WaitGroup{}
+	wg.Add(1)
+
+	go func() {
+		defer wg.Done()
+		startStandaloneNode(t, ctx, count, logMessages)
+	}()
+
+	_, err := waitForLogSubstring(ctx, logMessages, "started RPC server")
+	require.NoError(t, err, "RPC server didn't start")
+
+	client, err := rpc.NewClient(standaloneRPCEndpoint + strconv.Itoa(rpcPort+count))
+	require.NoError(t, err)
+
+	orderEventChan := make(chan []*zeroex.OrderEvent)
+	clientSubscription, err := client.SubscribeToOrders(ctx, orderEventChan)
+	require.NoError(t, err)
+	assert.NotNil(t, clientSubscription, "clientSubscription not nil")
+
+	ethClient := ethclient.NewClient(ethRPCClient)
+	signedTestOrder := scenario.CreateZRXForWETHSignedTestOrder(t, ethClient, makerAddress, takerAddress, wethAmount, zrxAmount)
+	expectedOrderHash, err := signedTestOrder.ComputeOrderHash()
+	require.NoError(t, err, "could not compute order hash for standalone order")
+
+	expectedFillableTakerAssetAmount := signedTestOrder.TakerAssetAmount
+	signedTestOrders := []*zeroex.SignedOrder{signedTestOrder}
+	_, err = client.AddOrders(signedTestOrders)
+	require.NoError(t, err)
+
+	// Reset the hash so that the order events can be compared without needing to hash
+	// the response order.
+	signedTestOrder.ResetHash()
+
+	orderEvent := <-orderEventChan
```

Comment:
Can we use a timeout here via a `select` statement? Otherwise if something goes wrong this test will hang until the test run times out (currently set to 30s).

You can look at [another part of the code where we do this](https://github.com/0xProject/0x-mesh/blob/017d42d5a46e7376a7e3bb8879a59b095ca8772c/ethereum/blockwatch/block_watcher_test.go#L60) for an example.

Reply:
I added a `context.WithTimeout` to this test. This should kill the test after 10 seconds. I'm going to go ahead and also add the `select`, but this seems redundant. What do you think?